### ISSUE TITLE
Run ansible-lint --profile min on a role won't report `parser-error` and `syntax-check`

### ISSUE THREAD

**Issue by kostyaplis** [ansible/ansible-lint#2891]
##### Summary
Not sure if it is a bug or by design.
There is a scenario where I only want to check the role just for basic syntax and parser errors (i.e. profile: min)
If profile: min failed -> interrupt the pipeline, no reason to continue with test deployment that will eventually fail
If profile: min succeed -> continue with extensive checks and test deployment using the role

It seems that `parser-error` and `syntax-check` rules are not applied when running 
```console
ansible-lint --profile min path/to/role
```

I understand that I can just remove `--profile min` check and let it fail with `--profile basic`, but then I confused with profile min purpose. 
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 6.10.2 using ansible 2.14.1
```

##### STEPS TO REPRODUCE

Doesn't fail on a role
```yaml
# role/tasks/main.yml
- name: test
```

```console
$ ansible-lint --profile min role/ -vvv
DEBUG    Logging initialized to level 10
....
DEBUG    Activating rule `internal-error` due to profile `min`
DEBUG    Activating rule `load-failure` due to profile `min`
DEBUG    Activating rule `parser-error` due to profile `min`
DEBUG    Activating rule `syntax-check` due to profile `min`
....
DEBUG    4/50 rules included in the profile
....
Passed with production profile: 0 failure(s), 0 warning(s) on 2 files.
```

But fails on a playbook
```yaml
# playbook-test.yml
-tasks:
  - name: test
```

```console
$ ansible-lint --profile min playbook-test.yml -vvv
....
syntax-check[specific]: no module/action detected in task.
playbook-test.yml:2:5
....
Failed after : 1 failure(s), 0 warning(s) on 1 files.
```

##### Desired Behavior

Should it fail on a role as well??

**Comment by ssbarnea** [comment]
@kostyaplis I confirm the bug and I will work to address it this week. It is a bit of a weird one but that does not make it less important.

**Comment by ajinkyau** [comment]
fixed as part of #3272
